### ISSUE TITLE
Negative weights producing a mesh of 1's and flat FFTPower at 0

### ISSUE THREAD

**Issue by safonova** [bccp/nbodykit#620]
I am working on calculating the power spectrum with FFTPower. When I create a mesh with a negative-valued weight column in an ArrayCatalog, the mesh preview ends up returning a grid filled with 1s. The power spectrum we get from the FFTPower['power'].real comprises zeros and nan values. The FFTPower['power'].real works fine with weights that are the same, but positive.

Here is the MWE. `z0_positions` is an array of 3d positions.  `mesh_one` is a mesh created without weights.
```
values = np.ones(len(z0_positions))
weights = -0.01*np.ones(len(z0_positions))
catalog_dictionary = {"Position": z0_positions, "Weight": weights, "Value": values}
catalog = ArrayCatalog(catalog_dictionary)

mesh_result = catalog.to_mesh(Nmesh=mesh_size,
                              position='Position',
                              compensated=True,
                              window='tsc',
                              value='Value',
                              weight='Weight',
                              BoxSize=[boxsize, boxsize, boxsize])

mesh_preview = mesh_result.preview(Nmesh=8)  # This returns a grid filled with "1."
FFT_result = FFTPower(first=mesh_one,  # mesh without weights
                      second=mesh_result, mode='1d',
                      dk=0.000005,
                      kmin=0.001*1e-3,
                      kmax=4, 
                      poles=[0],
                      BoxSize=[boxsize, boxsize, boxsize])

Pk_result = FFT_result.power 
Pk = Pk_result['power'].real  # This is a flat line at 0.
```

**Actual behaviour**
The mesh preview ends up returning a grid filled with 1s. The power spectrum we get from the FFTPower['power'].real comprises zeros and nan values. 

**Expected behaviour**
This should return a nonzero power spectrum. The mesh preview should be filled with the delta values scaled by the weight applied to the mesh.

**Version**
0.3.13 of nbodykit

**Comment by rainwoodman** [comment]
This is interesting.  I am curious about the use case though, as we usually do not expect weight to be negative -- I picture it as the effective number of particles -- and that's typically positive definite.

Based on this interpretation, if all particles are weighted the same, then they are effectively unweighed -- we normalize by the sum of weight, that's how you 1s on the grid, which has no power. 

The behaviour you expected seems to be consistent to setting Value? (still, a strictly uniform field has no power).

**Comment by bscot** [comment]
@rainwoodman I also ran into this issue and was hoping for a fix or workaround. The use case that I was considering is a flux weighted field where a continuum had been fit and subtracted, causing spurious negative flux values that carry information about the systematic I am studying.

As an aside, since this should be the same for all observations at a given redshift, it would be nice to be able to "paint" the value of the flux directly on the mesh since it is solely a function of frequency/redshift. I didn't see this use case in the cookbook - is there a simple way to do this?

**Comment by rainwoodman** [comment]
This is tricky -- the weight is noisy, and we probably want a MLE estimation of the painted field instead.

I also remember seeing terms like \sum w_i^2 / (\sum w_i)^2. So a simpler question to consider is, how do we extend this term when w_i is noisy. w_i = w~_i + n_i.

The true weight w~_i = w_i - n is not drawn from a gaussian, because it must be positive. (noise n is a gaussian I suppose). If we write down (a reasonable guess) the distribution of w~_i, and try to build a MLE estimator of \sum w_i^2 / (\sum w_i)^2 by marginalizing n and averaging over the pdf of w~_i, then perhaps we can get some inspiration how to extend the weighting scheme for noisy weights?

**Comment by bscot** [comment]
@rainwoodman I agree that the approach you suggest seems like the most intuitive extension. This seems related to the MLE problem with censored data, e.g. https://data.princeton.edu/wws509/notes/c7s2
